### PR TITLE
Make sure we aren't paying negative amounts to a Nation bank account during tax time.

### LINKED ISSUE #6540: Nation percentage tax bugged, taking negative taxes & bankrupting nations
### What steps will reproduce the problem?

1. Set nation percentage tax such as 3%
2. Have at least one town in the nation that is bankrupt and reaches bankruptcy cap & deletion?
3. Execute New Day
4. The nation attempts to tax negative values and somehow ends up in the negative amounts by way bigger numbers
5. Nation ends up deleted due to having negative money

### What is the expected output?

Nation taking 3% from positive balances normally. Bankrupt towns should be fully ignored by the tax if bankrupt towns paying tax is set to false, but they are not.

It seems that a town that was about to be deleted due to reaching bankruptcy cap was taxed with nation tax, even though this is supposed to be disabled by the config. (**do_bankrupt_towns_pay_nation_tax: 'false'**)

The town had negative balance and tax was somehow still applied, this town's name in this case was Albuquerque and it seems to have taxed in alphabetical order and towns after Albuquerque had negative taxes involved or somehow messed up if they were not able to pay the full cap. (max_nation_tax_percent_amount: '250')

We've had this issue before and in that case a bankrupt town was involved as well. There seem to have been other cases in which minor taxes have been applied as negative even if towns were not getting deleted by bankruptcy. Those are listed in the money.log.

This is now the 2nd case where we've had a major nation randomly fall due to nation % taxes. I reported this issue before but nothing was found wrong back then. We ran our own tests before enabling nation % taxes this time and it worked fine, but now when bankrupt towns are thrown into the mix, it still goes badly wrong.

### Towny version

0.98.6.7

### Server version

Paper version git-Paper-81 (MC: 1.19)

### Please use Pastebin.com to link the following files

1. money.log https://pastebin.com/gKDE0X3a
2. latest.log https://pastebin.com/ksdGttYk
3. Related configs https://pastebin.com/WYyg4FY6

### PR DESCRIPTION
**Pull request by LlmDl**
#### Description: 

Bankrupt towns' balances would return a negative amount when a nation had % tax enabled.

This called the payTo() function with a negative amount for the collector which then hits Vault with a negative deposit (which we are warned not to do.)

Solution:
Don't pay a nation tax if the amount is less than or equal to zero.

Also added is an override to BankAcount for canPayFromHoldings() that tests if a town is able to pay with debt.

____
#### New Nodes/Commands/ConfigOptions: 

____
#### Relevant Towny Issue ticket:

Closes #6540.
____
- [ ] I have tested this pull request for defects on a server. 

By making this pull request, I represent that I have the right to waive copyright and related rights to my contribution, and agree that all copyright and related rights in my contributions are waived, and I acknowledge that the TownyAdvanced organization has the copyright to use and modify my contribution under the Towny [License](https://github.com/LlmDl/Towny/blob/master/LICENSE.md) for perpetuity.

### CODE REVIEW

**Comment by LlmDl** [comment]
@Caleston I was unable to reproduce what you saw but it could be eco plugin related, you did not supply a startup.
If you'd like to test a jar I can get it to you.

**Comment by Caleston** [comment]
> @Caleston I was unable to reproduce what you saw but it could be eco plugin related, you did not supply a startup. If you'd like to test a jar I can get it to you.

Startup eco & Towny: https://pastebin.com/5tdNqs38
Using iConomy v5.14 (Towny Edition) on MYSQL.

I can test I suppose, I was about to update to latest version anyway.

File: src/com/palmergames/bukkit/towny/object/economy/BankAccount.java [hunk 1]
```diff
@@ -111,6 +111,14 @@ protected boolean addMoney(double amount) {
 		return TownyEconomyHandler.add(getName(), amount, world);
 	}
 
+	@Override
+	public boolean canPayFromHoldings(double amount) {
+		if (isBankrupt())
+			return addDebt(amount);
```
Comment (Warriorrrr) [inline src/com/palmergames/bukkit/towny/object/economy/BankAccount.java:117]: This should use the getDebtCap() instead of directly adding debt

**Review by Warriorrrr — CHANGES_REQUESTED** [review]

File: src/com/palmergames/bukkit/towny/object/economy/BankAccount.java (hunk 1 above)
Comment (LlmDl) [reply src/com/palmergames/bukkit/towny/object/economy/BankAccount.java]: Good catch.

**Review by Warriorrrr — APPROVED** [review]

**Pull request merged**